reconcile/external_resources/reconciler.py

```diff
@@ -70,10 +70,14 @@ class ReconciliationK8sJob(K8sJob, BaseModel, frozen=True):
     dry_run_suffix: str = ""
 
     def name_prefix(self) -> str:
+        identifier = (
+            f"{self.reconciliation.key.provider}-{self.reconciliation.key.identifier}"
+        )
         if self.is_dry_run:
-            return f"er-dry-run-mr-{self.dry_run_suffix}"
+            # dry-run-suffix: gitlabmergeid, <7 chars in the foreseeable future
+            return f"er-dry-run-{identifier[:45]}-{self.dry_run_suffix[:7]}"
         else:
-            return "er"
+            return f"er-{identifier[:53]}"
```

Comment:
Can you write a more detailed comment about where they come the slice values? They seem rather magical ATM.

Reply:
~~And those will cut the name automatically - so we maybe dont really have to take the effort to truncate on our own - we could just re-order, to make sure the suffix id is unique~~
If its applied directly and not via controller, then there will be no auto truncate and we have to handle it